app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/Model/DataSet.kt

```diff
@@ -0,0 +1,40 @@
+package com.github.HumanLearning2021.HumanLearningApp.Model
+
+
+/**
+ * A class representing a data set of images
+ *
+ * @param name the name of the data set
+ * @param admin the admin who created the data set
+ * @param categories the set of categories of images present in the dataset
+ * @param comVoorLevel the ComVoor level (1-5) describing the difficulty of the data set
+ */
+abstract class DataSet(
+    val name: String,
+    val admin: Admin,
+    val categories: Set<Category>,
+    val comVoorLevel: Int
+) {
+
+    /**
+     * A function to retrieve a picture from the data set given a category
+     *
+     * @param category the category of the image to be retrieved
+     */
+    abstract fun getPicture(category: Category): CategorizedPicture
```

Comment:
```suggestion
    abstract fun getPicture(category: Category): Task<CategorizedPicture>
```

This one needs to return an asynchronous task (`com.google.android.gms.tasks.Task`) Dependency is `    implementation 'com.google.android.gms:play-services-tasks:17.2.1'`. (Alternatively, tasks are isomorphic to Kotlin coroutines if you prefer that)